### ISSUE TITLE
ESLint LSP fails to load in Zed

### ISSUE THREAD

**Issue by 799189288** [zed-industries/zed#19709]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Language server error: eslint

oneshot canceled
-- stderr--
node:internal/modules/cjs/loader:1222
  throw err;
  ^

Error: Cannot find module 'C:\Users\79918\AppData\Local\Zed\languages\eslint\vscode-eslint-2.4.4\vscode-eslint\server\out\eslintServer.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1219:15)
    at Module._load (node:internal/modules/cjs/loader:1045:27)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:215:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:158:5)
    at node:internal/main/run_main_module:30:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.5.1

i checked my file folder, there is no 'out' in vscode-eslint\server

### Environment

windows 11 powershell

### If applicable, add mockups / screenshots to help explain present your vision of the feature
![40c61128-d1ee-4898-b729-ae7d206baa6f](https://github.com/user-attachments/assets/7495b8fd-3a97-41ee-91be-b784fb9ff61d)

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by kineticVis** [comment]
Having the same issue on Mac

**Comment by JulianCataldo** [comment]
Same on `macOS 14.7 23H124 x86_64` and Zed `Zed 0.159.6`.

```
Language server error: eslint

oneshot canceled
-- stderr--
node:internal/modules/cjs/loader:1252
  throw err;
  ^

Error: Cannot find module '/Users/***/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1249:15)
    at Function._load (node:internal/modules/cjs/loader:1075:27)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.10.0

```

**Comment by begprod** [comment]
Same Zed 0.159.7
This error make Zed useless

**Comment by begprod** [comment]
Fix in macos
`cd Users/${your_user_name}/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint`
`npm run webpack`

**Comment by akasakakona** [comment]
Same problem on Linux. There's nothing under `/vscode-eslint-2.4.4`. It's empty...

**Comment by FleetAdmiralJakob** [comment]
> Same problem on Fedora 41, Zed 0.162.5. There's nothing under `vscode-eslint-2.4.4`. It's empty.
> 
> Edit: I found the solution. Check your `vscode-eslint-2.4.4` folder. If it's empty then this can be solved fairly easily.
> 
> Upon reading the log, it seems like it failed to fetch the eslint module because this link is just broken: `https://github.com/microsoft/vscode-eslint/archive/refs/tags/release%2F2.4.4.tar.gz`
> 
> ```
> 2024-11-26T02:37:26.376128555-07:00 [INFO] downloading language server "eslint"
> 2024-11-26T02:37:27.809737266-07:00 [INFO] failed to fetch newest version of language server LanguageServerName("eslint"). error: extracting https://github.com/microsoft/vscode-eslint/archive/refs/tags/release%2F2.4.4.tar.gz to "/home/***/.local/share/zed/languages/eslint/vscode-eslint-2.4.4"
> ```
> 
> So what you need to do is to go to this link instead: [codeload.github.com/microsoft/vscode-eslint/tar.gz/release/2.4.4](https://codeload.github.com/microsoft/vscode-eslint/tar.gz/release/2.4.4) and download the `tar.gz` file provided.
> 
> Go to `vscode-eslint-2.4.4` folder, and extract the content of the `tar.gz` file downloaded into it.
> 
> Rename the `vscode-eslint-release-2.4.4` folder as `vscode-eslint`, `cd` into it, and run the following commands: `npm i` `npm audit fix` `npm run webpack` and that should fix it.

Works even if your folder isn't empty, just delete the folder and paste the new one in there

**Comment by beniaminzagan** [comment]
Hey guys, can you retry with Zed Preview 175.3 or a the latest stable? This might have been fixed already.

**Comment by stavalfi** [comment]
> Hey guys, can you retry with Zed Preview 175.3 or a the latest stable? This might have been fixed already.

Sorry, it did not:
```
zed -v
Zed 0.175.3 – /Applications/Zed Preview.app
```
error:
```
Language server error: eslint

oneshot canceled
-- stderr--
node:internal/modules/cjs/loader:1252
  throw err;
  ^

Error: Cannot find module '/Users/stava/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1249:15)
    at Function._load (node:internal/modules/cjs/loader:1075:27)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
    at node:internal/main/run_main_module:36:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v22.10.0
```

**Comment by beniaminzagan** [comment]
> > Hey guys, can you retry with Zed Preview 175.3 or a the latest stable? This might have been fixed already.
> 
> Sorry, it did not:
> 
> ```
> zed -v
> Zed 0.175.3 – /Applications/Zed Preview.app
> ```
> 
> error:
> 
> ```
> Language server error: eslint
> 
> oneshot canceled
> -- stderr--
> node:internal/modules/cjs/loader:1252
>   throw err;
>   ^
> 
> Error: Cannot find module '/Users/stava/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
>     at Function._resolveFilename (node:internal/modules/cjs/loader:1249:15)
>     at Function._load (node:internal/modules/cjs/loader:1075:27)
>     at TracingChannel.traceSync (node:diagnostics_channel:315:14)
>     at wrapModuleLoad (node:internal/modules/cjs/loader:218:24)
>     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)
>     at node:internal/main/run_main_module:36:49 {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: []
> }
> 
> Node.js v22.10.0
> ```

Hi, thank you for your repsonse, what happens if you delete thev vscode-eslint-2.4.4 folder at ~/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4, and restart zed?

**Comment by begprod** [comment]
> Hey guys, can you retry with Zed Preview 175.3 or a the latest stable? This might have been fixed already.

Hi, i install from scratch Zed 0.174.7, and again get the error

```js
Language server error: eslint

oneshot canceled
-- stderr--
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/Users/feelsbadfella/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.17.1
```

**Comment by andreasvh-conceto** [comment]
Hi @devzeth, 

also on my side in a typescript project, i got the same error in zed `0.174.7` after saving:

```typescript
Language server error: eslint

oneshot canceled
-- stderr--
node:internal/modules/cjs/loader:1228
  throw err;
  ^

Error: Cannot find module '/Users/.../Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.18.0

```

My settings:
```json
{
  "telemetry": {
    "metrics": false
  },
  "ui_font_size": 16,
  "buffer_font_size": 16,
  "vim_mode": true,
  "use_smartcase_search": true,
  "vim": {
    "toggle_relative_line_numbers": true,
    "highlight_on_yank_duration": 500,
    "use_smartcase_find": true
  },
  "outline_panel": {
    "default_width": 500
  },
  "toggle_relative_line_numbers": true,
  "theme": {
    "mode": "dark",
    "light": "One Light",
    "dark": "One Dark"
  },
  "inlay_hints": {
    "enabled": true
  },
  "format_on_save": "language_server",
  "terminal": {
    "font_family": "MesloLGS Nerd Font"
  },
  "languages": {
    "JavaScript": {
      "format_on_save": "language_server",
      "code_actions_on_format": {
        "source.fixAll.eslint": true
      }
    },
    "TypeScript": {
      "format_on_save": "language_server",
      "code_actions_on_format": {
        "source.fixAll.eslint": true
      }
    }
  },
  "base_keymap": "JetBrains",
  "buffer_line_height": "comfortable",
  "confirm_quit": false
}
```

Would love to see a fix. For now i will try @begprod [workaround](https://github.com/zed-industries/zed/issues/19709#issuecomment-2455288447)

Thanks

**Comment by juanjlunar** [comment]
The eslint is downloading fine with the version 0.176.1. The issue is that the language is not being installed correctly. If you go to the /Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint path (Mac OS) and you run `npm install` and `npm run webpack`, the language works fine even with prettier auto format.

**Comment by harryxu** [comment]
> The eslint is downloading fine with the version 0.176.1. The issue is that the language is not being installed correctly. If you go to the /Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint path (Mac OS) and you run `npm install` and `npm run webpack`, the language works fine even with prettier auto format.

In addition to the path /Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint, I also ran npm install and npm run webpack in ~/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server, and then it worked fine.

**Comment by hotdogc1017** [comment]
The easiest way is to delete zed's eslint. Take my installation path as an example
```bash
~/.local/share/zed/languages/eslint
```
Then delete the eslint folder and all its contents

```bash
rm -rf ~/.local/share/zed/languages/eslint
```

Then restart zed, and you can see the progress of eslint being reinstalled at the bottom of the editor and the eslint LSP runs successfully

**Comment by begprod** [comment]
@Angelk90 Hi, i'm not sure, i just fixed it like this and continue use zed. 
https://github.com/zed-industries/zed/issues/19709#issuecomment-2455288447

**Comment by gosvig123** [comment]
I'm seeing the same issue as the original poster, but on macOS.

**Error:**
```
Language server eslint:

initializing server eslint, id 3
-- stderr--
node:internal/modules/cjs/loader:1051
  throw err;
  ^

Error: Cannot find module '/Users/kristian/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.9.0
```

**Environment:**
- OS: macOS (darwin)
- Node: v20.9.0

**Troubleshooting Steps Taken:**
- Removed `~/Library/Application Support/Zed/languages/eslint` and restarted Zed.
- Ran `npm install` in the project directory.
- Ran `npm install eslint` in the project directory.
- Manually created the `.../server/out/` directory structure.

None of these steps resolved the issue. It seems the `eslintServer.js` file is not being created during the language server installation process. This appears to be related to the recent `vscode-eslint` version bump, as others have mentioned.

**Comment by fritx** [comment]
One of my Mac device has this trouble while the other works well.

Finally solve this nightmare:

```sh
zed -v 
Zed 0.194.3 – /Applications/Zed.app   (it is latest) 

rm -rf ~/Library/Application\ Support/Zed/languages/eslint
rm -rf ~/Library/Application\ Support/Zed/languages/vtsls

# and restart Zed
```

**Comment by fustosgergo** [comment]
**TL;DR:**  
If you get `Cannot find module .../Zed/server/out/eslintServer.js` when running ESLint in Zed, it’s probably because the `out/` folder isn’t built. Run `npm run webpack` inside the ESLint server folder (`.../Zed/server/`) to generate it.

---

Hi everyone,

I’ve been running into this issue on my Mac ever since I first downloaded Zed about half a year ago. I tried every suggestion I could find: removing the ESLint language server, deleting everything in `~/Library/Application Support/Zed`, even uninstalling and reinstalling Zed, but nothing worked.

But then I carefully read the error message:

`Error: Cannot find module ‘…/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js’`

When I checked the folder, I saw that there was no `out/` directory at all. The source code for the server was present in `server/`, but it hadn’t been built. Looking into `package.json`, I noticed there was a script named `webpack` that should build the server.

So I ran this inside the server directory:

```bash
cd ~/Library/Application\ Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server
npm run webpack
```

After reopening Zed, everything worked perfectly.

Hopefully this helps anyone else running into the same problem!

**Comment by noppa** [comment]
I faced this today on Linux. After some investigation, I think I found the cause (at least for my case).

If you have globally [disabled npm install scripts](https://docs.npmjs.com/cli/v7/using-npm/config#ignore-scripts), the ESLint plugin doesn't run its install scripts, which I assume should run the "webpack" build script.

You can check if you have the scripts disabled by running npm config get ignore-scripts -g or by opening $HOME/.npmrc.  
Like mentioned earlier in this thread, the fix is to either go to the ESLint plugin folder and manually run npm run webpack, or temporarily enable npm install scripts before opening Zed the next time.

I think this setting will (hopefully) become more and more popular with the recent major npm hacks, so it might be worth figuring out if Zed can still make extensions work out of the box with that setting enabled.

**Comment by GabrielBoehme13** [comment]
I'm also affected by this on Mac (since 0.192.5, and still now under 0.204.4). The cause is indeed as @noppa described – I've disabled NPM install scripts for security reasons. (…based on advice originally given in a talk by @rtfeldman, before he became part of the Zed team!)

For me, Zed was able to run ESLint just fine until the change from version 0.191.x to 0.192.5, which included a (since reverted) transition from vscode-eslint-2.4.4 to vscode-eslint-3.0.10. While trying to troubleshoot the resulting problems, I deleted my (presumably long-cached) "out/" directory, which explains why it hasn't worked for me since.

Unfortunately, trying to follow the advice given by @fustosgergo didn't work for me – `npm run webpack` failed with a bunch of compilation errors (specifically, 44 reported TypeScript errors), and didn't produce the missing "out/" directory. I'm not sure why it fails on my machine while running successfully for others, but this inconsistency is another reason to avoid having the ESLint Language Server installed this way by Zed.

I'd like to reiterate the last paragraph from @noppa – please find a way to get ESLint to work with Zed without requiring an NPM install script to perform webpack compilation on the end-user's machine. On some computers, we now know this can run into at least two failure states:

1. Failure due to NPM install scripts being disabled by the user
2. Failure due to `npm run webpack` encountering errors on the end-user's machine, resulting in no compiled output being generated

Additionally, both of these failures can be obscured due to old "out/" directories potentially remaining present after later failed compilation steps. This seems to have been the case on my machine prior to 0.192.5, and is potentially part of why the transition to vscode-eslint-3.0.10 encountered so many odd problems and had to be rolled back (some users continued to run old cached "out/" code that had been previously generated for vscode-eslint-2.*).

**Comment by kczx3** [comment]
Same on Linux

**Comment by gilmar-sales** [comment]
Tried to acces the Zed/languages/eslint/vscode-eslint-2.4.4/**vscode-eslint**, but found  Zed/languages/eslint/vscode-eslint-2.4.4/**vscode-eslint-release-2.4.4** instead

**Comment by gilmar-sales** [comment]
Running npm run webpack fails with `TS2307: Cannot find module './shared/settings'` the shared folder was **$shared** it builded after rename to **shared** and paste on server/src and client/src. Seems to be an folder symlink problem

**Comment by gilmar-sales** [comment]
After build, I removed the suffix  "-release-2.4.4" and eslint worked

**Comment by michaelw85** [comment]
I'm just giving zed a try, and this is the first issue I encounter, trying to build gives me:

```
ERROR in /Users/.../Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/client/tsconfig.json
/Users/.../Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/client/tsconfig.json
[tsl] ERROR
      TS2688: Cannot find type definition file for 'minimatch'.
  The file is in the program because:
    Entry point for implicit type library 'minimatch'
ts-loader-default_02839d45964937a4

webpack 5.102.1 compiled with 1 error in 3271 ms
```
Node v22.20.0
NPM 10.9.3

**Comment by GabrielBoehme13** [comment]
No changes for me under macOS and Zed 0.211.4 since [my last post](https://github.com/zed-industries/zed/issues/19709#issuecomment-3315492618) here.

It's now been 20 Zed minor versions since the last time I was able to use ESLint with Zed (0.191.x, ~4 months ago).

```
Language server eslint:

initializing server eslint, id 5: Server reset the connection
-- stderr --
node:internal/modules/cjs/loader:1423
  throw err;
  ^

Error: Cannot find module '/Users/--------/Library/Application Support/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out/eslintServer.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1420:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1058:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1063:22)
    at Module._load (node:internal/modules/cjs/loader:1226:37)
    at TracingChannel.traceSync (node:diagnostics_channel:328:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:244:24)
    at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:154:5)
    at node:internal/main/run_main_module:33:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v24.11.0

```

**Comment by weihsth** [comment]
EDITED ANSWER:
Just read the answer of @noppa and can confirm that I had `ignore-scripts=true` under $HOME/.npmrc.

So to not globally change this (not secure!), I tried this:
(This did not work!)

1. Added an .npmrc to my project with `ignore-scripts=false`
2. Closed Zed
3. Removed folder `[...]/Zed/languages/eslint`
4. Started Zed again

Since this did not change anything, I edited the global .npmrc under $HOME/.npmrc and did the rest of the steps above. That worked. But Zed team should make it possible to allow to overwrite .npmrc locally/per project.

For now I set back global config to ignore-scripts=true again after Zed installed eslint successfully.

---
OLD ANSWER:
Same here on Windows 11. There is no `[...]/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/out` folder.

These steps worked for me:
1. Go to `[...]/Zed/languages/eslint/vscode-eslint-2.4.4/vscode-eslint/server/`
2. Run `npm i`
3. Run `npm run webpack`
4. Restart language server(s) in Zed IDE

But these steps should be executed by Zed automatically, should't they?

**Comment by noppa** [comment]
Does that webpack config do something platform-dependent? I don't know about VSCode/Zed extensions, but generally it's very uncommon for npm packages to even need building on the end user's machine. They are usually built by the pacakge author and just the pre-built artifacts are supposed to be distributed via npm.  
There might be something that I'm overlooking about how the extension packages work, but if it's just bunch of JS that we are talking about, it sure would be simpler if installation didn't require touching any webpack stuff. And also faster to install.

**Comment by GabrielBoehme13** [comment]
It seems like it's not just platform-dependent, but computer-configuration-dependent, since `npm run webpack` [fails on my Mac with 44 errors](https://github.com/zed-industries/zed/issues/19709#issuecomment-3315492618), yet apparently runs just fine for [at least one other person who is also on a Mac](https://github.com/zed-industries/zed/issues/19709#issuecomment-3195968819).

I agree that this seems like something that should be downloaded in a ready-to-go state, rather than requiring the end user's computer to be configured a certain way for a post-install compile script to run successfully. (This might also allow Zed to upgrade to vscode-eslint v3 again, [after having to revert to v2](https://github.com/zed-industries/zed/issues/33425).)

**Comment by yeskunall** [comment]
Hey everyone, just checking in -- are you guys still running into this issue? Make sure to read [this comment](https://github.com/zed-industries/zed/issues/19709#issuecomment-3315263225) and see if it’s applicable to you.

If not, I’ll be happy to take another look. But, in general, all Zed does here is delegate to `npm` to install the extension or LSP.

**Comment by gabrielcosi** [comment]
I'm on Zed 1.1.6 (macOS) and the issue is still present. Why is this marked as completed?

```
Zed 1.1.6 
Commit: 96962914296a22c22a7f226b5ae7de4dd0e34729
Version: 1.1.6+stable.265.96962914296a22c22a7f226b5ae7de4dd0e34729
```

**Comment by wpiekutowski** [comment]
With npm 12, scripts will be blocked by default: https://github.blog/changelog/2026-06-09-upcoming-breaking-changes-for-npm-v12/. 

This issue needs to be revisited.

**Comment by GabrielBoehme13** [comment]
This issue is still present for me in Zed v1.6.3.

It's now been almost a full year since I've been able to use ESLint with Zed – the regression introduced in v0.192.5 (#33425) has never gone away for me, [even when manually attempting to run the postinstall steps](https://github.com/zed-industries/zed/issues/19709#issuecomment-3823783851).

And with the changes coming in npm v12, it sounds like the issues with the LSP's postinstall script will finally have to be addressed.

Please reopen this issue. Thanks!
